### ISSUE TITLE
Doesn't work on IOS

### ISSUE THREAD

**Issue by Ivanko5417** [Jaaneek/useFilePicker#23]
During using your lib I've noticed that it doesn't work in IOS (13.7). The popup opens but after choosing the files nothing changed. I've tested it for a little bit and figured out that `change` event is not fired. After a quick search, I've found out that `change` is not fired if an input is not in DOM. Function `openFileDialog` creates the input but doesn't add it to DOM.
So, might be worth adding an argument like `addToDOM` or even do it by default as it very popular case in my view.

**Comment by Ivanko5417** [comment]
Hey @Jaaneek! 
Could you please take a look at the issue?

**Comment by Jaaneek** [comment]
Hi Man, i will look into it after the weekend. Thank you for pointing that out and sorry for late response.

**Comment by Ivanko5417** [comment]
Hello @Jaaneek!
any updates?

**Comment by Ivanko5417** [comment]
Hey @Jaaneek!
I've opened a pull request with adding the input to the body to make it work on IOS. (https://github.com/Jaaneek/useFilePicker/pull/24)

**Comment by Jaaneek** [comment]
@MrKampla and I are investigating this, I didn't know of this issue when creating this package and I think putting the element into the document is something I would like to avoid. But for now I don't see a better option. I need to wait on @MrKampla opinion on this matter as well, and then we are ready to go

**Comment by MrKampla** [comment]
Hi @Ivanko5417, we've been investigating the issue and as @Jaaneek said, adding the element to the document is not considered as a good practice, so we'd like to avoid it. I've done some research, and it seems like ios does not always fire change events on inputs: https://stackoverflow.com/questions/8004227/ios-select-onchange-not-firing/10544704#10544704
My suggestion is to try adding onBlur event listener on top of existing onChange listener.
In src/helpers/openFileDialog.ts add:

> // set onchange event to call callback when user has selected file
>inputElement.addEventListener('change', callback);
>// fire a blur event to call a callback on ios devices, as they do not fire onChange event
>inputElement.addEventListener('blur', callback);

Test cases to cover:
- try to open a file selector, then close it without choosing any files
- check if change and blur events interfere with each other in any way on non-ios devices
- blur event callback should be called only on ios devices
- it would be nice to test it on React Native as well

If my solution with blur event doesn't work or introduces any new bugs, then we'll accept Your PR.

**Comment by Jaaneek** [comment]
@Ivanko5417 We opened another pull request that aims at fixing this issue, can you test it?
#25

**Comment by Ivanko5417** [comment]
Thank you very much for your PR. I've tested it and looks like it works!

**Comment by abFx** [comment]
For me I still see the issue while using it on Safari browser, the on change event is not always triggered. I need to try to upload a couple of times to make it work

**Comment by Ivanko5417** [comment]
What version of the library do you use?

**Comment by NicoMilk** [comment]
Hi !
Sorry to bump this, but I encounter the exact same problem : nothing happens after image selection.
With about all returns logged (filesContent, loading, errors and plainFiles),  console shows 0 value change
It works fine on Safari desktop

use-file-picker 1.4.1
iOS 13.3

**Comment by MrKampla** [comment]
@NicoMilk Can you show us which props are You passing to the useFilePicker hook? I've tested several scenarios, both on MacBook and iPhone, and I see that Safari doesn't like getting image files to be read as text - the browser just freezes on my devices. Try to disable file content reading with `readFileContent: false` and this should fix the problem. Everything works on Google Chrome without crashing, but I don't see a point in reading an image file as a text.

**Comment by NicoMilk** [comment]
@MrKampla, thanks for replying

The current config is :
`  // file picker config
  const [
    openFileSelector,
    { filesContent, loading, errors, plainFiles, clear },
  ] = useFilePicker({
    multiple: allowMultiple,
    readAs: 'DataURL', // availible formats: "Text" | "BinaryString" | "ArrayBuffer" | "DataURL"
    accept: ['.jpeg', '.jpg', '.png'],
    limitFilesConfig: limitFilesConfig,
    // minFileSize: 1, // in megabytes
    // maxFileSize: 1,
    // readFilesContent: false, // ignores file content
  })
`
with `allowMultiple` and `limitFilesConfig` set respectively to `false` and `{ min: 1, max: 3 }` by default
Again, the problem occurs only on Safari mobile. All our other tests (desktop Safari, Fifrefox, Chrome and mobile Firefox on Android) are running fine.
I'm gonna try to set `readFileContent: false` as you suggested and I'll let you know.

**Comment by NicoMilk** [comment]
Nothing changes with `readFileContent: false`

**Comment by Andriiklymiuk** [comment]
have anyone solved this issue? Also stuck on this one

**Comment by alessandro-bottamedi** [comment]
Same here, on safari mobile the change event does not fire...
